File: server/libs/os-services/data/service.js
const path = require('path');

const Program = require(path.join(__dirname, "program"));

const _state = Symbol('state');

const ServiceStates = {
    UNKNOWN: "unknown",
    RUNNING: "running",
    STOPPED: "stopped"
};

class Service extends Program {
    constructor(name, path) {
        super(name, path);
        this[_state] = ServiceStates.UNKNOWN;
    }

    static get States() { return ServiceStates; }

    get State() { return this[_state]; }
    set State(state) {
        if (state == ServiceStates.RUNNING || state == ServiceStates.STOPPED) {
            this[_state] = state;
        }
        else { this[_state] = ServiceStates.UNKNOWN; }
    }

    toJSON() {
        var json = super.toJSON();

        json.state = this.State;

        return json;
    }
}

module.exports = Service;

File: server/src/lib/os-utils/data/service.dart
import "./program.dart";

enum ServiceStatus { Unknown, Running, Stopped }

class Service extends Program {
  ServiceStatus _status = ServiceStatus.Unknown;

  Service(String name, String path) : super(name, path) {}

  ServiceStatus get status => this._status;
  void set status(ServiceStatus ss) {
    this._status = ss;
  }

  static ServiceStatus parseStatus(String sname) {
    if (sname == null) return ServiceStatus.Unknown;
    switch (sname.toLowerCase()) {
      // Windows service states
      case "stopped":
        return ServiceStatus.Stopped;
      case "running":
        return ServiceStatus.Running;
      case "unknown":
      default:
        return ServiceStatus.Unknown;
    }
  }
}
